backend/test/todos.test.ts

```diff
@@ -0,0 +1,287 @@
+            ? response.data.finishTodo.isDone
+            : undefined;
+        expect(isDone).toBe(true);
+      });
+      it('returns entire todo', () => {
+        const data = (response.data && response.data.finishTodo) || undefined;
+        expect(data).toMatchObject({
+          id: 'uuid',
+          description: 'test',
+          isDone: true,
+          assignedUser: null,
+        });
+      });
+    });
+  });
+
+  describe('deleteTodo', () => {
+    describe('given an ID', () => {
+      it('deletes a todo', async () => {
+        const mutation = gql`
+          mutation {
+            deleteTodo(todoId: "uuid") {
+              id
+              description
+              isDone
+              assignedUser {
+                name
+              }
+            }
+          }
+        `;
+
+        const response = await testClient.mutate({ mutation });
+        const data = (response.data && response.data.deleteTodo) || undefined;
+        expect(data).toMatchObject({
+          id: 'uuid',
+          description: 'test',
+          isDone: true,
+          assignedUser: null,
+        });
+      });
+
+      describe('buf if ID is invalid', () => {
+        it('returns `null`', async () => {
+          const mutation = gql`
+            mutation {
+              deleteTodo(todoId: "uuid") {
+                id
+                description
+                isDone
+                assignedUser {
+                  name
+                }
+              }
+            }
+          `;
+
+          const response = await testClient.mutate({ mutation });
+          const data = response.data || undefined;
+          expect(data).toMatchObject({
+            deleteTodo: null,
+          });
+        });
+      });
+    });
+  });
+});
+
+describe('query', () => {
+  describe('todos', () => {
+    it('returns a list of some todos', async () => {
+      const query = gql`
+        query {
+          todos {
+            description
+            isDone
+          }
+        }
+      `;
+
+      const response = await testClient.query({ query });
+      const data = response.data || undefined;
+      expect(data).toMatchObject({
+        todos: [
+          {
+            description: 'Number one',
+            isDone: true,
+          },
+          {
+            description: 'Number two',
+            isDone: false,
+          },
+        ],
+      });
+    });
+
+    describe('input arguments `options`', () => {
+      describe('(isDone: false)', () => {
+        it('returns unfinished todos only', async () => {
+          const query = gql`
+            query {
+              todos(options: { isDone: false }) {
+                description
+                isDone
+              }
+            }
+          `;
+
+          const response = await testClient.query({ query });
+          const data = response.data || undefined;
+          expect(data).toMatchObject({
```

Comment:
```suggestion
          await expect(testClient.query({ query })).resolves.toMatchObject({
```

Cool, huh?

http://favoritememes.com/_nw/44/87215078.jpg